assets/js/navigation.js

```diff
@@ -135,6 +147,32 @@
 			[].forEach.call( document.querySelectorAll( '.site-header ul ul, .site-header-cart .widget_shopping_cart' ), function( element ) {
 				element.classList.add( 'sub-menu--is-touch-device' );
 			} );
+
+			// Add blocked class to links that open sub-menus, and prevent from navigating away on first touch.
+			[].forEach.call( document.querySelectorAll( '.site-header .menu-item > a, .site-header .page_item > a, .site-header-cart a' ), function( anchor ) {
+				[ 'click', 'touchstart' ].forEach( function( event ) {
+					anchor.addEventListener( event, function( aEvent ) {
+						if ( 'click' !== aEvent.type || anchor.classList.contains( 'blocked' ) ) {
+							return true;
+						}
+
+						if ( ( 'cart-contents' === anchor.className && anchor.parentNode.nextElementSibling ) || anchor.nextElementSibling ) {
+							anchor.classList.add( 'blocked' );
+							aEvent.preventDefault();
+						}
+					} );
+				} );
+			} );
+
+			// Ensure the dropdowns close when user taps outside the site header
+			[].forEach.call( document.querySelectorAll( '.site-content, .header-widget-region, .site-footer' ), function( element ) {
```

Comment:
The dropdown does not close if the user clicks in the header area, e.g. to search products via the input, etc. It's not critical, but would be nice if that behaved in the same way as when clicking on somewhere else on the page.